api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -1084,4 +1088,75 @@ public OrderGroup getOrderGroup(Integer orderGroupId) throws APIException {
 		return dao.getOrderGroupsByEncounter(encounter);
 	}
 	
+	/**
+	 * @see org.openmrs.api.OrderService#getOrderGroupAttributeTypes()
+	 */
+	@Override
+	public List<OrderGroupAttributeType> getOrderGroupAttributeTypes() throws APIException {
+		return dao.getAllOrderGroupAttributeTypes();
+	}
+	
+	/**
+	 * @see org.openmrs.api.OrderService#getOrderGroupAttributeTypeById()
+	 */
+	@Override
+	public OrderGroupAttributeType getOrderGroupAttributeTypeById(Integer id) throws APIException {
+		return dao.getOrderGroupAttributeType(id);
+	}
+	
+	/**
+	 * @see org.openmrs.api.OrderService#getOrderGroupAttributeTypeByUuid()
+	 */
+	@Override
+	public OrderGroupAttributeType getOrderGroupAttributeTypeByUuid(String uuid)throws APIException {
+		return dao.getOrderGroupAttributeTypeByUuid(uuid);
+	}
+	
+	/**
+	 * @see org.openmrs.api.OrderService#saveOrderGroupAttributeType()
+	 */
+	@Override
+	public OrderGroupAttributeType saveOrderGroupAttributeType(OrderGroupAttributeType orderGroupAttributeType) throws APIException{
+		return dao.saveOrderGroupAttributeType(orderGroupAttributeType);
+	}
+
+	/**
+	 * @see org.openmrs.api.OrderService#retireOrderGroupAttributeType()
+	 */
+	@Override
+	public OrderGroupAttributeType retireOrderGroupAttributeType(OrderGroupAttributeType orderGroupAttributeType, String reason)throws APIException {
+		return dao.saveOrderGroupAttributeType(orderGroupAttributeType);
```

Comment:
Shouldn't this also use Context.getOrderService()?

Reply:
@dkayiwa  having replaced the dao variable with  for all the other methods drove me into a stackoverflow , for all the methods were the change was introduced.

https://hastebin.com/mupazayuko.js

however checking  applicationContect-service.xml , orderService is registered.

Any thoughts on the cause of this recursive behavour.

Reply:
I mean't only for this one. Not for any other method.